web/client/epics/wfsquery.js

```diff
@@ -153,11 +192,11 @@ export const wfsQueryEpic = (action$, store) =>
                 ...queryOptions
             };
             return Rx.Observable.merge(
-                getLayerJSONFeature({...layer, search: {...layer.search, url}}, ogcFilter, options)
+                (typeof filterObj === 'object' && getJSONFeatureWA(searchUrl, action.filterObj, options, selectedLayer) || getLayerJSONFeature({...layer, search: {...layer.search, url}}, ogcFilter, options))
                     .map(data => querySearchResponse(data, action.searchUrl, action.filterObj, action.queryOptions, action.reason))
                     .catch(error => Rx.Observable.of(queryError(error)))
                     .startWith(featureLoading(true))
-                    .concat(Rx.Observable.of(featureLoading(false)))
+                    .concat(Rx.Observable.of(featureLoading(false)), Rx.Observable.of(layerLoad(layer.id)))
```

Comment:
I don't understand the reason of these changes.  
I think you can avoid almost for sure the first row. Not sure about the `layerLoad` event at the end of the second changed row.

Reply:
the line 195 is changed because the getJSONFeatureWA signature from observables/wfs.js is changed => we add a parameter layer (set to selectedLayer in this call) which is used into getJSONFeature methods.
The original code was build this way with a call to getJSONFeatureWA and getLayerJSONFeature. We just add the new parameter

the line 199 is changed to make the layer loaded when it was a vector layer. This help to take into accound filter and so on.

Reply:
You should use instread the new method, to keep the things in sync. Can you use the new code?